### PR TITLE
bot commands from chat API

### PR DESCRIPTION
**Pull request by mmaxim**
Add support for advertising and listing bot commands. We do the following:

1. Provide new JSON API commands for advertising, clearing, and listing bot commands.
2. Attach bot commands to `InboxUIItem` for consumption by the UI.
3. Add interface for updating commands using the `GetBotInfo` RPC on the server.

### CODE REVIEW

File: go/chat/bots/commands.go [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+	Username      string
+}
+
+type commandsStorage struct {
+	Advertisements []storageCommandAdvertisement `codec:"A"`
+}
+
+const commandsStorageVersion = 1
+
+var commandsPublicTopicName = "___keybase_botcommands_public"
```
Comment (joshblum) [inline go/chat/bots/commands.go]: `const`?

File: go/chat/server.go [hunk 2]
```diff
@@ -3003,3 +3004,57 @@ func (h *Server) LocationDenied(ctx context.Context, convID chat1.ConversationID
...
+	defer func() { h.setResultRateLimit(ctx, &res) }()
+	_, err = utils.AssertLoggedInUID(ctx, h.G())
+	if err != nil {
+		return res, err
+	}
+	completeCh, err := h.G().BotCommandManager.UpdateCommands(ctx, convID, nil)
+	if err != nil {
+		return res, err
+	}
+	if err := <-completeCh; err != nil {
```
Comment (joshblum) [inline go/chat/server.go]: why block on this?

File: go/chat/bots/commands.go (hunk 1 above)
Comment (mmaxim) [reply go/chat/bots/commands.go]: I take it's address at some point so I made it this.

File: go/chat/server.go (hunk 2 above)
Comment (mmaxim) [reply go/chat/server.go]: Wanted the RPC to always return up to date results.

**Pull request merged**